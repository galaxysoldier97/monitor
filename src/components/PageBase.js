import React from 'react';
import { Divider, Paper } from '@material-ui/core';
import { KeyboardBackspace } from '@material-ui/icons';
import globalStyles from '../css/styles';
import { withRouter } from 'react-router-dom';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';

const GoBack = ({history}) => <KeyboardBackspace onClick={() => history.goBack()} alt="Go back" />;

GoBack.propTypes = {
  history: PropTypes.object,
};

const BackButton = withRouter(GoBack);

const PageBase = props => {
  const {title, navigation, backButton, subMenu, actionButton, children} = props;

  return (
    <>
      <Grid container justify="space-between">
        <Grid item>
          <Grid container spacing={1}>
            <Grid item style={{cursor: 'pointer'}}>
              {backButton && <BackButton />}
            </Grid>
            <Grid item style={globalStyles.navigation}>
              <Typography>{navigation}</Typography>
            </Grid>
          </Grid>
        </Grid>
        <Grid item>{subMenu}</Grid>
      </Grid>
      <Paper style={globalStyles.paper}>
        <Grid container justify="space-between" alignContent="center">
          <Grid item>
            <Typography variant="h3" style={globalStyles.title}>{title}</Typography>
          </Grid>
          <Grid item>
            {actionButton}
          </Grid>
        </Grid>
        <Divider />
        {children}
        <div style={globalStyles.clear} />
      </Paper>
    </>
  );
};

PageBase.propTypes = {
  title: PropTypes.string,
  navigation: PropTypes.string,
  children: PropTypes.element,
  backButton: PropTypes.bool,
  subMenu: PropTypes.element,
  actionButton: PropTypes.element,
};

export default PageBase;
