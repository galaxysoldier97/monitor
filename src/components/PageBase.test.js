import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import PageBase from './PageBase';
import Button from '@material-ui/core/Button';
import MenuItem from '@material-ui/core/MenuItem';

test('Page base snapshot', () => {
  const component = renderer.create(
    <BrowserRouter>
      <PageBase title="test" children={<div>test</div>} backButton actionButton={<Button>button</Button>} subMenu={<MenuItem>test</MenuItem>}/>
    </BrowserRouter>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
