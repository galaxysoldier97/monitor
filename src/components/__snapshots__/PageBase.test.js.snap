// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page base snapshot 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <svg
            alt="Go back"
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
            />
          </svg>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
          style={
            Object {
              "paddingBottom": 15,
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <li
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        test
      </li>
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "padding": 30,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-content-xs-center MuiGrid-justify-xs-space-between"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3"
          style={
            Object {
              "fontSize": 24,
              "fontWeight": undefined,
              "marginBottom": 20,
            }
          }
        >
          test
        </h3>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            button
          </span>
        </button>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div>
      test
    </div>
    <div
      style={
        Object {
          "clear": "both",
        }
      }
    />
  </div>,
]
`;
